### PR TITLE
designation-delete-api

### PR DESCRIPTION
**Pull request by Mathuragini**

### CODE REVIEW

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java [hunk 1]
```diff
@@ -2,7 +2,51 @@
...
+
+	@Autowired
+	private Mapper mapper;
+
+	private static final Logger logger = Logger.getLogger(DesignationController.class);
+
+	// delete api for designation
+	@DeleteMapping(value = EndpointURI.DELETE_DESIGNATION_BY_ID)
+	public ResponseEntity<Object> deleteDesignation(@PathVariable(Constants.DESIGNATION_PATH_VARIABLE) Long id) {
+		if (designationService.isDesignationIdAlreadyExist(id)) {
```
Comment (SSuja) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java]: change isDesignationIdAlreadyExist to isDesignationIdExist

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java [hunk 2]
```diff
@@ -2,7 +2,51 @@
...
+			designationService.deleteDesignation(id);
```
Comment (SSuja) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java]: What will happen for the employees who are having this Designation? Don't we need to check that here?

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/DesignationService.java [hunk 3]
```diff
@@ -2,4 +2,8 @@
 
 public interface DesignationService {
 
+	public boolean isDesignationIdAlreadyExist(Long id);
```
Comment (SSuja) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/DesignationService.java]: This is not to check id exist, this is to check isDesignationExist.So, name it as "isDesignationExist"

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java [hunk 4]
```diff
@@ -2,7 +2,51 @@
...
+	ValidationFailureStatusCodes validationFailureStatusCodes;
+
+	@Autowired
+	private Mapper mapper;
+
+	private static final Logger logger = Logger.getLogger(DesignationController.class);
+
+	// delete api for designation
+	@DeleteMapping(value = EndpointURI.DELETE_DESIGNATION_BY_ID)
+	public ResponseEntity<Object> deleteDesignation(@PathVariable(Constants.DESIGNATION_PATH_VARIABLE) Long id) {
```
Comment (SSuja) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java]: what is the purpose of having constants here like @PathVariable(Constants.DESIGNATION_PATH_VARIABLE), you can just implement like this: @PathVariable long id

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java (hunk 1 above)
Comment (Mathuragini) [reply tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java]: fixed

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/DesignationService.java (hunk 3 above)
Comment (Mathuragini) [reply tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/DesignationService.java]: fixed

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java (hunk 4 above)
Comment (Mathuragini) [reply tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/DesignationController.java]: fixed

**Pull request merged**